### PR TITLE
Docs: highlight structure vision

### PR DESCRIPTION
**Pull request by hsjobeki**
## Description of changes

- Better highlight the vision to improve guidance on possible contributors.
- Use more active wording to highlight the ongoing work.
- Encourage to write `doc-comments` since they can be retrieved with `:doc` or `nixdoc` which means we can generate them from source.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: doc/README.md [hunk 1]
```diff
@@ -2,10 +2,18 @@
...
 This directory houses the sources files for the Nixpkgs reference manual.
 
-Going forward, it should only contain [reference](https://nix.dev/contributing/documentation/diataxis#reference) documentation.
-For tutorials, guides and explanations, contribute to <https://nix.dev/> instead.
-
-For documentation only relevant for contributors, use Markdown files and code comments in the source code.
+> [!IMPORTANT]
+> We are actively restructuring our documentation to follow the [Diátaxis framework](https://diataxis.fr/)
+>
+> Going forward, this folder should only contain **[reference](https://nix.dev/contributing/documentation/diataxis#reference) documentation**.
```
Comment (fricklerhandwerk) [inline doc/README.md:8]:
```suggestion
> Going forward, this directory should **only** contain [reference documentation](https://nix.dev/contributing/documentation/diataxis#reference).
```

File: doc/README.md [hunk 2]
```diff
@@ -2,10 +2,18 @@
...
+> For tutorials, guides and explanations, contribute to <https://nix.dev/> instead.
+>
+> We are actively working to generate **all** reference documentation from the [doc-comments](https://github.com/NixOS/rfcs/blob/master/rfcs/0145-doc-strings.md) present in code.
+> This also provides the benefit of using `:doc` in the `nix repl` to view reference documentation locally on the fly.
+
+For reference documentation prefer Markdown files and code comments in the source code if supported yet.
```
Comment (fricklerhandwerk) [inline doc/README.md:14]: This has changed meaning, but I don't understand why.

```suggestion
For documentation only relevant for contributors, use Markdown files next to the source and regular code comments.
```

File: doc/README.md [hunk 3]
```diff
@@ -2,10 +2,18 @@
...
+
+> We'are actively soliciting feedback for missing support. If your case is not supported open an issue starting with `Doc:`
```
Comment (fricklerhandwerk) [inline doc/README.md:16]: I'm not sure what this trying to say. Support for what? Which case?

**Review by fricklerhandwerk — COMMENTED** [review]
That's a good thing to do! Just not sure about the details.

File: doc/README.md (hunk 2 above)
Comment (hsjobeki) [reply doc/README.md:14]: Doc-comments are not only relevant for contributors. They are reference documentation which is usually public.

Markdown files next to .nix files could be used if doc-comments directly in the nix source would be too big.
The nixdoc support is ready for this: https://github.com/nix-community/nixdoc/pull/114
But we didn't merge it. I guess we have a bootstrapping problem then.

Comment (fricklerhandwerk) [reply doc/README.md:14]: But then I don't understand the new sentence at all. What else should people use for documentation except markdown and source comments? The original sentence was a hint for separating end-user and contributor documentation, so it's unrelated to the highlighted note on using doc-comments, which is why I don't think it should even be touched.

File: doc/README.md (hunk 3 above)
Comment (fricklerhandwerk) [reply doc/README.md:16]: Something like this?

```suggestion
> [!TIP]
> Feedback for improving support for parsing and rendering doc-comments is highly appreciated.
> [Open an issue](https://github.com/NixOS/nixpkgs/issues/new?labels=6.topic%3A+documentation&title=Doc%3A+) to request bugfixes or new features.
```



Would be better if you created an issue template where you ask for what you need to know directly, then link it here.

File: doc/README.md (hunk 2 above)
Comment (hsjobeki) [reply doc/README.md:14]: Okay seems fine

**Review by fricklerhandwerk — APPROVED** [review]
You can do the issue template later and fix the link. :)

**Pull request merged**